test/unit-test-example.js

```diff
@@ -9,6 +9,17 @@ class ExampleTest extends BaseTest {
   testInvalidCall () {
     expect(() => Example.broken()).toThrow()
   }
+
+  async testCatchError () {
+    const shouldThrow = () => {
+      throw new Error('hello!')
+    }
+    const error = await this.catchError(shouldThrow)
+    expect(error.message).toEqual('hello!')
+    const doesNotThrow = () => 2
+    await expect(this.catchError(doesNotThrow))
+      .rejects.toThrow()
+  }
```

Comment:
Looks good, but this test doesn't show case the intended use case. Can you update the test case to make sure it allows you to check other properties on the error?

Also wondering if there is an easy way to extent expect itself through injection or something else, so the expect API works better off shelf.
```
await expect(method()).toThrow(e => e.status == 200 && e.message.contains('abc'))
```

Reply:
Yes to the first part. To the second part, that is a very interesting idea, but may not play well with my true use case;
For SQL validation, I want to provide error-by-line data, which will require a deep comparison of properties on the returned error. `.toThrow(e => e.status == 200 && e.message.contains('abc'))` wouldn't work well for that case, in that the jest matcher would just fail without showing me a diff of the desired properties.

Reply:
updated test case to validate returned error more thoroughly. Feel free to reopen conversation to discuss second part (extending `.toThrow`)